deployments/main.tf

```diff
@@ -37,13 +37,19 @@ module "dpe-sandbox-spacelift-development" {
   cluster_name = "dpe-k8-sandbox"
   vpc_name     = "dpe-sandbox"
 
-  vpc_cidr_block                         = "10.52.16.0/20"
-  public_subnet_cidrs                    = ["10.52.16.0/24", "10.52.17.0/24"]
+  vpc_cidr_block = "10.52.16.0/20"
+  # A public subnet is required for each AZ in which the worker nodes are deployed
+  public_subnet_cidrs                    = ["10.52.16.0/24", "10.52.17.0/24", "10.52.19.0/24"]
   private_subnet_cidrs_eks_control_plane = ["10.52.18.0/28", "10.52.18.16/28"]
   azs_eks_control_plane                  = ["us-east-1a", "us-east-1b"]
 
-  private_subnet_cidrs_eks_worker_nodes = ["10.52.20.0/22", "10.52.24.0/22", "10.52.28.0/22"]
-  azs_eks_worker_nodes                  = ["us-east-1a", "us-east-1b", "us-east-1c"]
+  private_subnet_cidrs_eks_worker_nodes = ["10.52.28.0/22", "10.52.24.0/22", "10.52.20.0/22"]
+  azs_eks_worker_nodes                  = ["us-east-1c", "us-east-1b", "us-east-1a"]
```

Comment:
The juggling of this is an unfortunate side affect of how the VPC module creates subnets in specific AZs. By doing this it allows us to:

    Create 3 public subnets in each AZ
    Create 2 private subnets in a, and b for the eks control plane
    Create 3 private subnets in each AZ for the private worker nodes.

Why this is needed:

    In order for the AWS load balancer to forward traffic to a private subnet, there needs to be a public subnet in that AZ.
    Worker nodes can be deployed to all 3 AZs, and because of this when a pod was scheduled into the c subnet, the load balancer could not forward traffic to that instance.